table/tables/tables.go

```diff
@@ -345,15 +345,9 @@ func (t *TableCommon) UpdateRecord(ctx context.Context, sctx sessionctx.Context,
 	sh := memBuffer.Staging()
 	defer memBuffer.Cleanup(sh)
 
```

Comment:
Also, need to handle delete statements?

Here is an example:
1. prepare, execute in session 1.
```sql
create table t (a bigint, b int, index idx(a));
insert into t values (1,1);
 alter table t modify a int unsigned;
```
2. execute in session 2 and the DDL job state in `StateWriteReorganization`, I got below error:

```sql
test> delete from t where a=1;
(8039, 'Index column _Col$_a offset out of bound, offset: 2, row: [KindInt64 1 KindInt64 1]')
```